game/src/contrib/systems/HealthSystem.java

```diff
@@ -34,22 +35,44 @@ public void execute() {
                 .map(this::applyDamage)
                 // Filter all dead entities
                 .filter(hsd -> hsd.hc.isDead())
-                .filter(
-                        hsd -> {
-                            if (!hsd.ac.hasAnimation(AdditionalAnimations.DIE)
-                                    || hsd.ac
-                                            .getAnimation(AdditionalAnimations.DIE)
-                                            .get()
-                                            .isLooping()) return true;
-                            if (!hsd.ac.isCurrentAnimation(AdditionalAnimations.DIE)) {
-                                hsd.ac.currentAnimation(AdditionalAnimations.DIE);
-                            }
-                            return hsd.ac.currentAnimation().isFinished();
-                        })
+                // Set DeathAnimation if possible and not yet set
+                .map(this::triggerDeathAnimation)
+                .filter(this::filterByAnimation)
                 // Remove all dead entities
                 .forEach(this::removeDeadEntities);
     }
 
+    /**
+     * Filters the stream by checking the Animation of an Entity.
+     *
+     * @param hsd HSData to check Animations in
+     * @return true if Entity can be removed from the game
+     */
+    private boolean filterByAnimation(HSData hsd) {
+        // test if hsd has a DeathAnimation
+        Predicate<HSData> hasDeathAnimation =
+                (hsData) -> hsData.ac.hasAnimation(AdditionalAnimations.DIE);
+        // test if Animation is looping
+        Predicate<HSData> isAnimationLooping =
+                (hsData) -> hsData.ac.getAnimation(AdditionalAnimations.DIE).get().isLooping();
+        // test if Animation has finished playing
+        Predicate<HSData> isAnimationFinished =
+                (hsData) -> hsData.ac.currentAnimation().isFinished();
```

Comment:
@JudiTeller Finde ich recht übersichtlich! Drei Gedanken/Vorschläge:

(1) Mache die Prädikate bitte für die AnimationComponent, also `Predicate<AnimationComponent>`. Dann brauchst Du das nicht erst aus dem hsData jedesmal rausfischen. Entsprechend müssten die Aufrufe dann auch irgendwie so sein:

```java
AnimationComponent ac = hsd.ac;
return !hasDeathAnimation.test(ac)
                || isAnimationLooping.test(ac)
                || isAnimationFinished.test(ac);
```

(2) Sollte `isLooping()` nicht auch auf der `currentAnimation()` geprüft werden? Oder kann eine sonstige Animation auch gerade am Loopen sein? Also sollte `hsData.ac.getAnimation(AdditionalAnimations.DIE).get().isLooping();` eher ein `hsData.ac.currentAnimation().isLooping()` sein? (Vielleicht bin ich aber zu weit weg vom Thema.)

(3) Ich würde darüber nachdenken, ob `isLooping()` und `isFinished()` nicht Bestandteil von `AnimationComponent` werden sollten? Das müsste sich dort freilich auch die `currentAnimation()` beziehen. @AMatutat Was denkst Du?

Reply:
> (3) Ich würde darüber nachdenken, ob isLooping() und isFinished() nicht Bestandteil von AnimationComponent werden sollten? Das müsste sich dort freilich auch die currentAnimation() beziehen. @AMatutat Was denkst Du?

Klar warum nicht, ist zwar ein sehr starker edge-case aber was solls.
Im `DrawComponent` zwei Methoden hinzufügen:
- `isFinished` ->. `return currentAnimation.isFinished();`
- `isLooping` -> `return currentAnimation.isLooping();`

jetzt weiß ich grade nicht, ob eine Looping Animation bei `isFinished()` immer false zurückliefert. Das wäre zumindest meine erste Erwartung. Müsste man nochmal nachschauen.

Reply:
Würde für mich Sinn machen, dass "looping==true" auch "isFinished==false" liefert und umgekehrt. Dann könnte man die Logik hier sogar noch weiter zusammenziehen ...

Reply:
Zur klarstellung:
Wir unterschieden bei Animationen zwischen Looping-Animation und Non-Looping-Animation.

Eine Looping-Animation springt nachdem der letzte Frame abgespielt wurde, wieder zum Anfang zurück. Loopt also. Das sind 90% unserer Animationen. 

Non-Looping-Animationen bleiben beim letzten Frame stehen, springen also nicht mehr zurück zum ersten Frame. Der letzte Frame wird endlos angezeigt (oder bis die Animation gewechselt wird). Das macht z.B bei Schatzkisten sinn. Du öffnest die Schatzkiste und die Animation zeigt dir, wie die Kiste sich langsam öffnet. Wenn die Kiste ganz offen ist, bleibt sie natürlich offen.

Reply:
Hab die Methoden in DrawComponent implementiert